internal/pkg/cli/app_status.go

```diff
@@ -0,0 +1,363 @@
+	initDescriber func(*appStatusOpts, string) error
+}
+
+func newAppStatusOpts(vars appStatusVars) (*appStatusOpts, error) {
+	ssmStore, err := store.New()
+	if err != nil {
+		return nil, fmt.Errorf("connect to environment datastore: %w", err)
+	}
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, err
+	}
+
+	return &appStatusOpts{
+		appStatusVars: vars,
+		storeSvc:      ssmStore,
+		ws:            ws,
+		w:             log.OutputWriter,
+		initecsSvc: func(o *appStatusOpts, env *archer.Environment) error {
+			sess, err := session.NewProvider().FromRole(env.ManagerRoleARN, env.Region)
+			if err != nil {
+				return fmt.Errorf("session for role %s and region %s: %w", env.ManagerRoleARN, env.Region, err)
+			}
+			o.ecsSvc = ecs.New(sess)
+			return nil
+		},
+		initcwSvc: func(o *appStatusOpts, env *archer.Environment) error {
+			sess, err := session.NewProvider().FromRole(env.ManagerRoleARN, env.Region)
+			if err != nil {
+				return fmt.Errorf("session for role %s and region %s: %w", env.ManagerRoleARN, env.Region, err)
+			}
+			o.cwSvc = cloudwatch.New(sess)
+			return nil
+		},
+		initDescriber: func(o *appStatusOpts, appName string) error {
+			d, err := describe.NewWebAppDescriber(o.ProjectName(), appName)
+			if err != nil {
+				return fmt.Errorf("creating describer for application %s in project %s: %w", appName, o.ProjectName(), err)
+			}
+			o.describer = d
+			return nil
+		},
+	}, nil
+}
+
+// Validate returns an error if the values provided by the user are invalid.
+func (o *appStatusOpts) Validate() error {
+	if o.ProjectName() != "" {
+		if _, err := o.storeSvc.GetProject(o.ProjectName()); err != nil {
+			return err
+		}
+	}
+	if o.appName != "" {
+		if _, err := o.storeSvc.GetApplication(o.ProjectName(), o.appName); err != nil {
+			return err
+		}
+	}
+	if o.envName != "" {
+		if _, err := o.storeSvc.GetEnvironment(o.ProjectName(), o.envName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Ask asks for fields that are required but not passed in.
+func (o *appStatusOpts) Ask() error {
+	if err := o.askProject(); err != nil {
+		return err
+	}
+	return o.askAppEnvName()
+}
+
+// Execute shows the applications through the prompt.
+func (o *appStatusOpts) Execute() error {
+	if err := o.configSvc(); err != nil {
+		return err
+	}
+	serviceArn, err := o.describer.GetServiceArn(o.envName)
+	if err != nil {
+		return fmt.Errorf("get service ARN: %w", err)
+	}
+	clusterName, err := serviceArn.ClusterName()
+	if err != nil {
+		return fmt.Errorf("get cluster name: %w", err)
+	}
+	serviceName, err := serviceArn.ServiceName()
+	if err != nil {
+		return fmt.Errorf("get service name: %w", err)
+	}
+	service, err := o.ecsSvc.Service(clusterName, serviceName)
+	if err != nil {
+		return fmt.Errorf("get ECS service %s: %w", serviceName, err)
+	}
+	tasks, err := o.ecsSvc.ServiceTasks(clusterName, serviceName)
+	if err != nil {
+		return fmt.Errorf("get ECS tasks for service %s: %w", serviceName, err)
+	}
```

Comment:
I wonder if all these dependencies should be inside `WebAppStatus`? so that we have:
```go 
describer := describe.NewWebAppStatus(appName, envName, projectName)
description, err := describer.Describe()
if err != nil {
	return err
}
if o.shouldOutputJSON {
	data, err := appStatus.JSONString()
	if err != nil {
		return err
	}
	fmt.Fprintf(o.w, data)
} else {
	fmt.Fprintf(o.w, appStatus.HumanString())
}
```
This way execute becomes crystal clear

Reply:
Sounds good to me. I'll have a try